nucleus/deployment/common/src/main/java/com/sun/enterprise/deploy/shared/FileArchive.java

```diff
@@ -296,12 +303,19 @@ public boolean isDirectory(String name) {
      * @return an enumeration of the archive file entries.
      */
     @Override
-    public Enumeration<String> entries(String prefix) {
+    public Enumeration<String> entries(String prefix) {        
         prefix = prefix.replace('/', File.separatorChar);
         File file = new File(archive, prefix);
-        List<String> namesList = new ArrayList<>();
-        getListOfFiles(file, namesList, null);
-        return Collections.enumeration(namesList);
+        
+        if (entriesCache.containsKey(file.hashCode())) {
+            return entriesCache.get(file.hashCode());
+        } else {
+            List<String> namesList = new ArrayList<String>();
+            getListOfFiles(file, namesList, null);
+            Enumeration entries = Collections.enumeration(namesList);
+            entriesCache.put(file.hashCode(), entries);
```

Comment:
Looks like you are doing the job of a HashMap explicitly. Wouldn't it be sufficient do put the file in as key?